Error compiling on Ubuntu 32 bit

Which operating system does the issue occur on?
Ubuntu 18.04

If on linux, are you using X11 or Wayland?
X11

$ rustc --version
rustc 1.29.0 (aa3ca1994 2018-09-11)

And I have installed [extra libraries](https://github.com/jwilm/alacritty#debianubuntu-1).
```shell
 Running `rustc --crate-name alacritty src/lib.rs --crate-type lib --emit=dep-info,link -C opt-level=3 -C debuginfo=1 --cfg 'feature="default"' -C metadata=5ee2b18ca4c9524a -C extra-filename=-5ee2b18ca4c9524a --out-dir /home/dikiaap/.local/src/alacritty/target/release/deps -L dependency=/home/dikiaap/.local/src/alacritty/target/release/deps --extern arraydeque=/home/dikiaap/.local/src/alacritty/target/release/deps/libarraydeque-08ee37e97b90e418.rlib --extern base64=/home/dikiaap/.local/src/alacritty/target/release/deps/libbase64-e6eabaf53e7cc438.rlib --extern bitflags=/home/dikiaap/.local/src/alacritty/target/release/deps/libbitflags-b0c8b66076b97148.rlib --extern cgmath=/home/dikiaap/.local/src/alacritty/target/release/deps/libcgmath-0d8067f9268911f7.rlib --extern clap=/home/dikiaap/.local/src/alacritty/target/release/deps/libclap-f307362568949e59.rlib --extern copypasta=/home/dikiaap/.local/src/alacritty/target/release/deps/libcopypasta-58ada58588ce64e6.rlib --extern env_logger=/home/dikiaap/.local/src/alacritty/target/release/deps/libenv_logger-e6fd71e176733754.rlib  --extern errno=/home/dikiaap/.local/src/alacritty/target/release/deps/liberrno-6dfd0d62082eba63.rlib --extern fnv=/home/dikiaap/.local/src/alacritty/target/release/deps/libfnv-c908a0dca541167d.rlib --extern font=/home/dikiaap/.local/src/alacritty/target/release/deps/libfont-258430fbd2fcb0f6.rlib --extern glutin=/home/dikiaap/.local/src/alacritty/target/release/deps/libglutin-af341f60b8b6fa52.rlib --extern libc=/home/dikiaap/.local/src/alacritty/target/release/deps/liblibc-e3c5af6976bd5979.rlib --extern log=/home/dikiaap/.local/src/alacritty/target/release/deps/liblog-772ee7f320a2bec5.rlib --extern mio=/home/dikiaap/.local/src/alacritty/target/release/deps/libmio-2d18fd33b3c7d7e0.rlib --extern mio_more=/home/dikiaap/.local/src/alacritty/target/release/deps/libmio_more-684d665528f741d7.rlib --extern notify=/home/dikiaap/.local/src/alacritty/target/release/deps/libnotify-5951ca4b812defa2.rlib --extern parking_lot=/home/dikiaap/.local/src/alacritty/target/release/deps/libparking_lot-27fbe5ef26366209.rlib --extern serde=/home/dikiaap/.local/src/alacritty/target/release/deps/libserde-80a63e587b7907f5.rlib --extern serde_derive=/home/dikiaap/.local/src/alacritty/target/release/deps/libserde_derive-46c4171b1916aa33.so --extern serde_json=/home/dikiaap/.local/src/alacritty/target/release/deps/libserde_json-eefe52ccde8f9d85.rlib --extern serde_yaml=/home/dikiaap/.local/src/alacritty/target/release/deps/libserde_yaml-7b75d4a36d5d8bec.rlib --extern static_assertions=/home/dikiaap/.local/src/alacritty/target/release/deps/libstatic_assertions-ba749e4bb42cac79.rlib --extern unicode_width=/home/dikiaap/.local/src/alacritty/target/release/deps/libunicode_width-f17deb847f8d49c1.rlib --extern vte=/home/dikiaap/.local/src/alacritty/target/release/deps/libvte-c41d051770f3056c.rlib --extern x11_dl=/home/dikiaap/.local/src/alacritty/target/release/deps/libx11_dl-8c327cb4bfebebf0.rlib --extern xdg=/home/dikiaap/.local/src/alacritty/target/release/deps/libxdg-040b08a04276b707.rlib -L native=/usr/lib/i386-linux-gnu -L native=/usr/lib/i386-linux-gnu -L native=/usr/lib/i386-linux-gnu -L native=/usr/lib/i386-linux-gnu -L native=/home/dikiaap/.local/src/alacritty/target/release/build/libloading-4e622ceb383faa3e/out`

error[E0512]: transmute called with types of different sizes
   --> src/grid/storage.rs:226:9
    |
226 |         assert_eq_size!(Row<T>, [u32; 8]);
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: source type: grid::row::Row<T> (128 bits)
    = note: target type: [u32; 8] (256 bits)
    = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error: aborting due to previous error

For more information about this error, try `rustc --explain E0512`.
error: Could not compile `alacritty`.

Caused by:
  process didn't exit successfully: `rustc --crate-name alacritty src/lib.rs --crate-type lib --emit=dep-info,link -C opt-level=3 -C debuginfo=1 --cfg 'feature="default"' -C metadata=5ee2b18ca4c9524a -C extra-filename=-5ee2b18ca4c9524a --out-dir /home/dikiaap/.local/src/alacritty/target/release/deps -L dependency=/home/dikiaap/.local/src/alacritty/target/release/deps --extern arraydeque=/home/dikiaap/.local/src/alacritty/target/release/deps/libarraydeque-08ee37e97b90e418.rlib --extern base64=/home/dikiaap/.local/src/alacritty/target/release/deps/libbase64-e6eabaf53e7cc438.rlib --extern bitflags=/home/dikiaap/.local/src/alacritty/target/release/deps/libbitflags-b0c8b66076b97148.rlib --extern cgmath=/home/dikiaap/.local/src/alacritty/target/release/deps/libcgmath-0d8067f9268911f7.rlib --extern clap=/home/dikiaap/.local/src/alacritty/target/release/deps/libclap-f307362568949e59.rlib --extern copypasta=/home/dikiaap/.local/src/alacritty/target/release/deps/libcopypasta-58ada58588ce64e6.rlib --extern env_logger=/home/dikiaap/.local/src/alacritty/target/release/deps/libenv_logger-e6fd71e176733754.rlib --extern errno=/home/dikiaap/.local/src/alacritty/target/release/deps/liberrno-6dfd0d62082eba63.rlib --extern fnv=/home/dikiaap/.local/src/alacritty/target/release/deps/libfnv-c908a0dca541167d.rlib --extern font=/home/dikiaap/.local/src/alacritty/target/release/deps/libfont-258430fbd2fcb0f6.rlib  --extern glutin=/home/dikiaap/.local/src/alacritty/target/release/deps/libglutin-af341f60b8b6fa52.rlib --extern libc=/home/dikiaap/.local/src/alacritty/target/release/deps/liblibc-e3c5af6976bd5979.rlib --extern log=/home/dikiaap/.local/src/alacritty/target/release/deps/liblog-772ee7f320a2bec5.rlib --extern mio=/home/dikiaap/.local/src/alacritty/target/release/deps/libmio-2d18fd33b3c7d7e0.rlib --extern mio_more=/home/dikiaap/.local/src/alacritty/target/release/deps/libmio_more-684d665528f741d7.rlib --extern notify=/home/dikiaap/.local/src/alacritty/target/release/deps/libnotify-5951ca4b812defa2.rlib --extern parking_lot=/home/dikiaap/.local/src/alacritty/target/release/deps/libparking_lot-27fbe5ef26366209.rlib --extern serde=/home/dikiaap/.local/src/alacritty/target/release/deps/libserde-80a63e587b7907f5.rlib --extern serde_derive=/home/dikiaap/.local/src/alacritty/target/release/deps/libserde_derive-46c4171b1916aa33.so --extern serde_json=/home/dikiaap/.local/src/alacritty/target/release/deps/libserde_json-eefe52ccde8f9d85.rlib --extern serde_yaml=/home/dikiaap/.local/src/alacritty/target/release/deps/libserde_yaml-7b75d4a36d5d8bec.rlib --extern static_assertions=/home/dikiaap/.local/src/alacritty/target/release/deps/libstatic_assertions-ba749e4bb42cac79.rlib --extern unicode_width=/home/dikiaap/.local/src/alacritty/target/release/deps/libunicode_width-f17deb847f8d49c1.rlib --extern vte=/home/dikiaap/.local/src/alacritty/target/release/deps/libvte-c41d051770f3056c.rlib --extern x11_dl=/home/dikiaap/.local/src/alacritty/target/release/deps/libx11_dl-8c327cb4bfebebf0.rlib --extern xdg=/home/dikiaap/.local/src/alacritty/target/release/deps/libxdg-040b08a04276b707.rlib -L native=/usr/lib/i386-linux-gnu -L native=/usr/lib/i386-linux-gnu -L native=/usr/lib/i386-linux-gnu -L native=/usr/lib/i386-linux-gnu -L native=/home/dikiaap/.local/src/alacritty/target/release/build/libloading-4e622ceb383faa3e/out` (exit code: 1)
```

--- a/src/grid/storage.rs
+++ b/src/grid/storage.rs
@@ -223,7 +223,7 @@ impl<T> Storage<T> {
     /// instructions. This implementation achieves the swap in only 8 movups
     /// instructions.
     pub fn swap(&mut self, a: usize, b: usize) {
-        assert_eq_size!(Row<T>, [u32; 8]);
+        assert_eq_size!(Row<T>, [usize; 4]);
 
         let a = self.compute_index(a);
         let b = self.compute_index(b);
@@ -232,13 +232,13 @@ impl<T> Storage<T> {
             // Cast to a qword array to opt out of copy restrictions and avoid
             // drop hazards. Byte array is no good here since for whatever
             // reason LLVM won't optimized it.
-            let a_ptr = self.inner.as_mut_ptr().add(a) as *mut u64;
-            let b_ptr = self.inner.as_mut_ptr().add(b) as *mut u64;
+            let a_ptr = self.inner.as_mut_ptr().add(a) as *mut usize;
+            let b_ptr = self.inner.as_mut_ptr().add(b) as *mut usize;
 
             // Copy 1 qword at a time
             //
             // The optimizer unrolls this loop and vectorizes it.
-            let mut tmp: u64;
+            let mut tmp: usize;
             for i in 0..4 {
                 tmp = *a_ptr.offset(i);
                 *a_ptr.offset(i) = *b_ptr.offset(i);

